flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/deployment/AbstractDeploymentObserver.java

```diff
@@ -252,6 +257,60 @@ private void onMissingDeployment(FlinkDeployment deployment) {
                 EventUtils.Component.JobManagerDeployment);
     }
 
+    /**
+     * Checks a deployment that is currently in the UPGRADING state whether it was already deployed
+     * but we simply miss the status information. After comparing the target resource generation
+     * with the one from the possible deployment if they match we update the status to the already
+     * DEPLOYED state.
+     *
+     * @param flinkDep Flink resource to check.
+     * @param context Context for reconciliation.
+     */
+    private void checkIfAlreadyUpgraded(FlinkDeployment flinkDep, Context context) {
```

Comment:
How about calling this method as 'syncWithDeployment` or similar. How problematic it is to introduce a new DEPLOYING state?

Reply:
Well this check specifically checks if an UPGRADING deployment is already upgraded. Calling it sync would give the impression that it does something more.

Also it's not really problematic to introduce a new DEPLOYING state but it would add no value so I would prefer not do it.

Reply:
The way I understand the method catches up with initial deployments too, and the UPGRADING state covers both status. Distinguishing two states DEPLOYING and UPGRADING makes sense to me, or calling the method `checkIfAlreadyDeployedOrUpgraded`. But maybe I don't clearly understand the logic here.

Reply:
If I could I would simply rename UPGRADING -> DEPLOYING. But since that's not an option in this case I would prefer to simply keep it called UPGRADING and use that state also for initial first deployments.